chore: more utility clean up from bigtable-hbase module

rahulKQL opened the pull request:
This PR removes references of `BigtableInstanceName` and `ThreadUtil`, which resides in `bigtable-core` module. ThreadUtil is copied over to `bigtable-hbase` from the core module.

rahulKQL committed:
chore: more utility clean up from bigtable-hbase module

This PR removes references of `BigtableInstanceName` and `ThreadUtil`, which resides in bigtable-core module. ThreadUtil is copied over to `bigtable-hbase`.

athakor commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/util/ThreadUtil.java:
```diff
@@ -0,0 +1,50 @@
+   * Get a {@link ThreadFactory} suitable for use in the current environment.
+   *
+   * @param nameFormat to apply to threads created by the factory.
+   * @param daemon {@code true} if the threads the factory creates are daemon threads, {@code false}
```
`if the threads the factory creates are daemon threads` this looks weird

rahulKQL replied:
Thanks for the review! Done.

dmitry-fa replied:
Get a ... => Get**s** a ... 
If class description stays:  "creates" may be this method should not get, but create?

rahulKQL replied:
Ack. Thanks for review!

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/util/ThreadUtil.java:
```diff
@@ -0,0 +1,50 @@
+ *
+ * <p>For internal use only - public for technical reasons.
+ */
+@InternalApi("For internal usage only")
```
Not a good practice to start class description with Creates

dmitry-fa replied:
Creates a ... => Utility class to create a ...

rahulKQL replied:
Thanks for suggestion!

dmitry-fa commented on bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/util/ThreadUtil.java:
```diff
@@ -0,0 +1,50 @@
+ *
+ * <p>This class copies code that originates in {@link
+ * io.grpc.internal.GrpcUtil#getThreadFactory(String, boolean)}.
+ *
```
Are you sure that this note should be here?   (I don't know, just asking)

rahulKQL replied:
Yes, It still holds true for our utility. So it's best to keep track of it.

rahulKQL committed:
chore: address feedback comments

 - used GrpcUtil#getThreadFactory for creating thread factory.
 - moved `ThreadUtil.java` under `src/test`

rahulKQL committed:
chore: fix dependency missing failure
